crates/ruma-common/src/identifiers/matrix_uri.rs

```diff
@@ -298,26 +298,34 @@ impl MatrixToUri {
 
     /// Try parsing a `&str` into a `MatrixToUri`.
     pub fn parse(s: &str) -> Result<Self, Error> {
-        let without_base = if let Some(stripped) = s.strip_prefix(MATRIX_TO_BASE_URL) {
-            stripped
-        } else {
-            return Err(MatrixToError::WrongBaseUrl.into());
-        };
-
-        let url = Url::parse(MATRIX_TO_BASE_URL.trim_end_matches("#/"))
-            .expect("matrix.to base URL is valid")
-            .join(without_base)
-            .map_err(|_| MatrixToError::InvalidUrl)?;
-
-        let id = MatrixId::parse_with_sigil(url.path())?;
-        let mut via = vec![];
-
-        for (key, value) in url.query_pairs() {
-            if key.as_ref() == "via" {
-                via.push(value.parse()?);
-            } else {
-                return Err(MatrixToError::UnknownArgument.into());
-            }
+        let s = s.strip_prefix(MATRIX_TO_BASE_URL).ok_or(MatrixToError::WrongBaseUrl)?;
+        let s = s.strip_suffix('/').unwrap_or(s);
+
+        // Separate the identifiers and the query.
```

Comment:
```suggestion
        // Separate the identifiers and the query.
        //
        // We do this instead of Url::parse, as canonically # comes after ?,
        // matrix URIs often arent.
```

Reply:
I added a comment at the beginning of the method explaining why we don't use `Url`